service/s3/api_op_PutObject.go

```diff
@@ -399,3 +400,42 @@ func newServiceMetadataMiddleware_opPutObject(region string) *awsmiddleware.Regi
 		OperationName: "PutObject",
 	}
 }
+
+// PresignPutObject is used to generate a presigned HTTP Request which contains
+// presigned URL, signed headers and HTTP method used.
+func (c *PresignClient) PresignPutObject(ctx context.Context, params *PutObjectInput, optFns ...func(*PresignOptions)) (*v4.PresignedHTTPRequest, error) {
+	if params == nil {
+		params = &PutObjectInput{}
+	}
+	var presignOptions PresignOptions
+	for _, fn := range optFns {
+		fn(&presignOptions)
+	}
+	if len(optFns) != 0 {
+		c = NewPresignClient(c.client, optFns...)
+	}
+
+	clientOptFns := make([]func(o *Options), 0)
+	clientOptFns = append(clientOptFns, func(o *Options) {
+		o.HTTPClient = &smithyhttp.NopClient{}
+	})
+
+	ctx = presignedurlcust.WithIsPresigning(ctx)
+	result, _, err := c.client.invokeOperation(ctx, "PutObject", params, clientOptFns,
+		addOperationPutObjectMiddlewares,
+		c.convertToPresignMiddleware,
+		addPutObjectPayloadAsUnsigned,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	out := result.(*v4.PresignedHTTPRequest)
+	return out, nil
+}
+
+func addPutObjectPayloadAsUnsigned(stack *middleware.Stack, options Options) error {
+	v4.RemoveContentSHA256HeaderMiddleware(stack)
```

Comment:
This header middleware is probably still useful. just don't need the compute sha. Doesn't the unsigned payload middleware expect the content sha header middleware to be present to set the header?

Reply:
Looks like i was wrong on this one, it should be fine to remove this middleware.